turbopack/crates/turbo-tasks-fs/src/rope.rs

```diff
@@ -375,12 +375,20 @@ pub mod ser_as_string {
 impl PartialEq for Rope {
     // Ropes with similar contents are equals, regardless of their structure.
     fn eq(&self, other: &Self) -> bool {
-        if Arc::ptr_eq(&self.data, &other.data) {
-            return true;
-        }
         if self.len() != other.len() {
             return false;
         }
+        self.cmp(other) == Ordering::Equal
+    }
+}
+
+impl Eq for Rope {}
+
+impl Ord for Rope {
```

Comment:
There's some unit tests for equality at the bottom of this file. One or two unit tests for ordering (nothing fancy) would be nice.